### ISSUE TITLE
Order of execution issue with postgresql_default_auth_method: "md5" and user creation

### ISSUE THREAD

**Issue by morgangraphics** [ANXS/postgresql#459]
Setting the **postgresql_default_auth_method: "md5"** on a fresh install tells postgres that the default postgres user requires password authentication without first having set up an initial password before hand. This is a order of execution issue where:

1. the configuration files are provisioned (essentially requiring passwords)
2. users are attempting to be created but passwords are not set 

Ideally the order should be 
1. user creation should happen first
2. the configurations applied afterwards locking up the database access.  

Minimum test case for generating issue

```
  - role: ansible-role-postgres
        postgresql_version: 12
        postgresql_default_auth_method: "md5"
        postgresql_users:
          - name: postgres
            pass: somegeneratedpassword123
            encrypted: yes
```

current work around is to accept the default **postgresql_default_auth_method: "peer"**

and after the posgres role has run,  edit the /etc/postgresql/12/main/pg_hba.conf file 

from 
```
# local postgres without password
local  all  postgres    peer
```

to

```
# local postgres without password
local  all  postgres    md5
```

in another task.

**Comment by morgangraphics** [comment]
In thinking about this further. Perhaps it might be a little easier for the role to have a default `pg_hba.conf`  file added before user creation which allows for unobstructed user creation and then the `pg_hba.conf` configurations are applied after user creation is successful.

**Comment by gclough** [comment]
@morgangraphics, can you just use the `postgresql_pg_hba_default` parameter to include what you want?

```
  postgresql_pg_hba_default:
    - comment: "Local postgresql Unix user"
      type: local
      database: all
      user: postgres
      address: ""
      method: "md5"
```

**Comment by morgangraphics** [comment]
I'm no longer involved in the project where this came up. If the above proposal this really the "correct" way to solve this issue, then I think the only thing would be to update the documentation accordingly pointing out that like in #473 order is important.
